easybuild/tools/downloadtools.py

```diff
@@ -0,0 +1,187 @@
+##
+# Copyright 2015 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+Set of tools to download files.
+@author: Jens Timmerman
+"""
+import os
+import shutil
+import sys
+import urllib2
+
+from vsc.utils import fancylogger
+
+from easybuild.tools import build_log  # import build_log must stay, to activate use of EasyBuildLog
+from easybuild.tools.config import build_option
+from easybuild.tools.repository.svnrepo import SvnRepository
+from easybuild.tools.filetools import make_tarfile, write_file, mkdir
+
+
+_log = fancylogger.getLogger('filetools', fname=False)
+
+# list of possible protocols we should know how to download
+HTTP = 'download_http'
+# GIT = download_git
+SVN = 'download_svn'
+# BZR = download_bzr
+FILE = 'download_local_file'
+# FTP = download_ftp
+
+# map protocols to things we know how to download
+PROTOCOL_MAP = {
+    'http': HTTP,
+    'https': HTTP,
+    # 'http+git': GIT,
+    # 'git': GIT,
+    # 'bzr': BZR,
+    # 'ftp': FTP,
+    'svn': SVN,
+    'svn+http': SVN,
+    'file': FILE,
```

Comment:
and don't include commented ones, it's OK if some of these yield `NotImplementedError` imho